frontends/p4/typeChecking/typeUnification.cpp

```diff
@@ -303,64 +302,142 @@ bool TypeUnification::unify(const IR::Node* errorPosition,
             return false;
         }
         auto ts = src->to<IR::Type_BaseList>();
-        if (td->components.size() != ts->components.size()) {
+        bool convenient;
```

Comment:
you have to initialize convenient. Doesn't the C compiler give you a warning about this?